src/topicmodeller/topicmodeller/topicmodeller.py

```diff
@@ -0,0 +1,155 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from boilerpipe.extract import Extractor
+from gensim import corpora, models
+from nltk.corpus import stopwords
+
+import codecs
+import datetime
+import jsonpickle
+import nltk
+import os
+import re
+
+
+class TopicModellerDocument(object):
+    def __init__(self, url, topics):
+        self.url = url
+        self.topics = topics
+
+
+class RepeatableBatchedDocuments(object):
+    def __init__(self, folder, batch_size):
+        self.folder = folder
+        self.batch_size = batch_size
+
+    def __iter__(self):
+        scraper_documents = []
+        documents = []
+        for file_name in os.listdir(self.folder):
+
+            file_content = open(os.path.join(self.folder, file_name)).read()
+            scraper_document = _decode(file_content)
+            readable_document = _readable_document(scraper_document.html_content)
+
+            scraper_documents.append(scraper_document)
+            documents.append(_clean(_word_tokenize(readable_document)))
+
+            if len(documents) == self.batch_size:
+                yield (scraper_documents, documents)
+                scraper_documents = []
+                documents = []
+        if len(documents) != 0:
+            yield (scraper_documents, documents)
+
```

Comment:
This batching method is used only for initialize_dictionnary but not for classify_and_dump_json (but in the second I suppose we also need the unitary treatment "clean/tokenize/decode/readable"). To get this treatment in classify_and_dump_json  we use RepeatableBatchedDocuments but we "unbatch it" (two for loops)

I think this is because we mix two separate things : batching (only for init_dico) and clean (no needed to use batching), this way we won't do the double for loop